golang/cosmos/x/vtransfer/keeper/keeper.go

```diff
@@ -199,21 +307,38 @@ func (k Keeper) InterceptOnTimeoutPacket(
 
 // InterceptWriteAcknowledgement checks to see if the packet's receiver is a
 // targeted account, and if so, delegates to the VM.
-func (k Keeper) InterceptWriteAcknowledgement(ctx sdk.Context, chanCap *capabilitytypes.Capability, packet ibcexported.PacketI, ack ibcexported.Acknowledgement) error {
+func (k Keeper) InterceptWriteAcknowledgement(ctx sdk.Context, chanCap *capabilitytypes.Capability, packet ibcexported.PacketI, ack ibcexported.Acknowledgement) (ibcexported.Acknowledgement, ibcexported.PacketI) {
 	// Get the base baseReceiver from the packet, without computing a stripped packet.
 	baseReceiver, err := types.ExtractBaseAddressFromPacket(k.cdc, packet, types.RoleReceiver, nil)
+
+	timeout := clienttypes.NewHeight(
+		packet.GetTimeoutHeight().GetRevisionNumber(),
+		packet.GetTimeoutHeight().GetRevisionHeight(),
+	)
+	origPacket := channeltypes.NewPacket(
+		packet.GetData(), packet.GetSequence(),
+		packet.GetSourcePort(), packet.GetSourceChannel(),
+		packet.GetDestPort(), packet.GetDestChannel(),
+		timeout, packet.GetTimeoutTimestamp(),
+	)
+	packetStore, packetKey := k.PacketStoreFromOrigin(ctx, types.PacketSrc, packet)
```

Comment:
If we're writing an ack, wouldn't we be the destination / receiver? Why are we using a src origin here?